packages/two_dimensional_scrollables/lib/src/table_view/table.dart

```diff
@@ -829,17 +982,35 @@ class RenderTableViewport extends RenderTwoDimensionalViewport {
       _paintCells(
         context: context,
         offset: offset,
-        leading: const TableVicinity(column: 0, row: 0),
-        trailing:
+        leadingVicinity: TableVicinity.zero,
+        trailingVicinity:
             TableVicinity(column: _lastPinnedColumn!, row: _lastPinnedRow!),
       );
     }
   }
 
+  @override
+  RenderBox? getChildFor(
+    ChildVicinity vicinity, {
+    bool allowMerged = true,
+  }) {
+    return super.getChildFor(vicinity) ??
+        (allowMerged ? _getMergedChildFor(vicinity as TableVicinity) : null);
```

Comment:
Actually, the `allowMerged` parameter is not super clear to me. Can you explain?

Reply:
Yes, I think that is nice and cleaner, thanks! 
The parameter is meant to distinguish between times when it should return null. I think it could use a better name, but I could not think of one.

Some examples... when resolving TableSpan decorations, and we are walking through unmerged cells to create the largest rect we can for the unmerged portion, returning null from getChildFor signals we have reached the end of the span, since merged cell vicinities that would return null are already parsed out.

The other case is when painting children. We iterate through all of the vicinities, and when null is returned for one, we can validate we have a merged cell that covers that area and move on to the next vicinity.

When allowMerged is true (default), it means, we can get the merged child that covers the given vicinity, regardless of which vicinity that child was actually assigned.

Reply:
Re: `allowMerged` - That makes sense. Thanks for the explanation. I think we should add a comment regarding that to the code here. A better name would also be desirable. Maybe something like `mapMergedVicinityToCanonicalChild` as in "if true, it will return the canonical child for the merged cell, if false, it will return whatever value in the underlying child data structure is".